### PR TITLE
Add docs about how import works in extensions and about testing

### PR DESCRIPTION
**Pull request by marekweb**
Currently there is no mention of how to test extensions, and no mention of `@sourcegraph/extension-api-stubs` in the docs.

There is also no explanation about how importing the `sourcegraph` module works -- namely that it's an empty module.

This adds pages to the extension docs to cover these topics.

### CODE REVIEW

File: doc/extensions/authoring/importing_sourcegraph.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+## How the import works
+
+Unlike a regular module that is installed with `npm` or `yarn` and is packaged
+or bundled with your project, the `sourcegraph` module is actually injected
+dynamically when your extension is loaded within a Sourcegraph instance.
+
+The `sourcegraph` module that you install as a `package.json` dependency is
+actually a set of Typescript type definitions that allow you to use the
+Sourcegraph API's interfaces. The module doesn't contain the implementations of
+these interfaces.
```
Comment (tjkandala) [inline doc/extensions/authoring/importing_sourcegraph.md:27]: **nit**: "Typescript" should be "TypeScript"

**thought**: Great explanation! I just want to make sure that a reader with no prior knowledge of the extension API will understand that they still need to import `sourcegraph` if they're writing their extension in JavaScript . You explicitly mention this in the previous section, so I don't think we need to reiterate it here. Should we conduct some user interviews once we're done with all documentation to see if users properly understand key points of the extension API?

**Review by tjkandala — APPROVED** [review]

File: doc/extensions/authoring/importing_sourcegraph.md (hunk 1 above)
Comment (marekweb) [reply doc/extensions/authoring/importing_sourcegraph.md:27]: Great suggestion, we really do need to get more feedback from users! And I think we will discover that we still have a lot of work to do to make the extension platform clearer, easier to understand, and better explained.

**Comment by felixfbecker** [comment]
Thought: If importing the `sourcegraph` module, and it being a "virtual" module is so confusing that we need a documentation page for it, why don't we just make `sourcegraph` a global similar to the `browser`/`chrome` object or `window`? It is provided by the environment after all, there can only be one version of it and it would make polyfilling it in tests easier.

**Review by felixfbecker — COMMENTED** [review]
Sorry I forgot to submit this, not blocking, just thoughts

**Pull request merged**